plugins/modules/oneview_server_profile.py

```diff
@@ -0,0 +1,674 @@
+        C(compliant) will make the server profile compliant with its server profile template, when this option was
+        specified. If there are Offline updates, the Server Hardware is turned off before remediate compliance issues
+        and turned on after that.
+    default: present
+    choices: ['present', 'absent', 'compliant']
+    type: str
+  data:
+    description:
+      - List with Server Profile properties.
+    required: true
+    type: dict
+  auto_assign_server_hardware:
+    description:
+      - Bool indicating whether or not a Server Hardware should be automatically retrieved and assigned to the Server Profile.
+        When set to true, creates and updates try to ensure that an available Server Hardware is assigned to the Server Profile.
+        When set to false, if no Server Hardware is specified during creation, the profile is created as 'unassigned'. If the
+        profile already has a Server Hardware assigned to it and a serverHardwareName or serverHardwareUri is specified as None,
+        the Server Profile will have its Server Hardware unassigned.
+    default: True
+    choices: [True, False]
+    type: bool
+  params:
+    description:
+      - Dict with query parameters.
+    required: False
+    type: dict
+notes:
+    - "For the following data, you can provide either a name or a URI: enclosureGroupName or enclosureGroupUri,
+       osDeploymentPlanName or osDeploymentPlanUri (on the osDeploymentSettings), networkName or networkUri (on the
+       connections list), volumeName or volumeUri (on the volumeAttachments list), volumeStoragePoolName or
+       volumeStoragePoolUri (on the volumeAttachments list), volumeStorageSystemName or volumeStorageSystemUri (on the
+       volumeAttachments list), serverHardwareTypeName or  serverHardwareTypeUri, enclosureName or enclosureUri,
+       firmwareBaselineName or firmwareBaselineUri (on the firmware), sasLogicalJBODName or sasLogicalJBODUri (on
+       the sasLogicalJBODs list) and initialScopeNames or initialScopeUris"
+    - "If you define the volumeUri as null in the volumeAttachments list, it will be understood that the volume
+       does not exist, so it will be created along with the server profile. Be warned that every time this option
+       is executed it will always be understood that a new volume needs to be created, so this will not be idempotent.
+       It is strongly recommended to ensure volumes with Ansible and then assign them to the desired server profile.
+       does not exists, so it will be created along with the server profile"
+
+extends_documentation_fragment:
+    - hpe.oneview.oneview
+    - hpe.oneview.oneview.validateetag
+    - hpe.oneview.oneview.params
+'''
+
+EXAMPLES = '''
+- name: Create a Server Profile from a Server Profile Template with automatically selected hardware
+  oneview_server_profile:
+    hostname: 172.16.101.48
+    username: administrator
+    password: my_password
+    api_version: 1600
```

Comment:
update api_versions to 2000 for all examples

Reply:
have updated. Perhaps, why on fc_network, storage_pool and many other files/main.yml shows API verison other than 2000.
We need to fix that too.